test/extended/router/http2.go

```diff
@@ -598,3 +623,22 @@ func platformHasHTTP2LoadBalancerService(platformType configv1.PlatformType) boo
 		return false
 	}
 }
+
+func isNoDNSZones(oc *exutil.CLI, timeout time.Duration) (bool, error) {
+	noDNSZones := false
+
+	if err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		dns, err := oc.AdminConfigClient().ConfigV1().DNSes().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			e2e.Logf("Get dnses.config/cluster failed: %v, retrying...", err)
+			return false, nil
+		}
+		if dns.Spec.PublicZone == nil && dns.Spec.PrivateZone == nil {
+			noDNSZones = true
```

Comment:
The condition `dns.Spec.PublicZone == nil && dns.Spec.PrivateZone == nil` is not available to check the `GCP user provisioned DNS` cluster, instead the cluster set `LoadBalancer.DNSManagementPolicy` as below
```
			if platformStatus.GCP != nil && platformStatus.GCP.CloudLoadBalancerConfig != nil &&
				platformStatus.GCP.CloudLoadBalancerConfig.DNSType == configv1.ClusterHostedDNSType {
				effectiveStrategy.LoadBalancer.DNSManagementPolicy = operatorv1.UnmanagedLoadBalancerDNS
			}
```

see https://github.com/openshift/cluster-ingress-operator/blob/cbc0b217b655f1f0ce0becc9145c2a6042beabea/pkg/operator/controller/ingress/controller.go#L478-L492

We might need to update the func to check default ingresscontroller OperatorCondition `Type:   "DNSManaged"` is `ConditionFalse`